### DISCUSSION TITLE
Improving Docker Image Update Efficiency with `latest` and `latest-rc` Tags

### DISCUSSION THREAD

**Discussion by jacobshilitz** [karrioapi/karrio#521] [General]
When we tag Docker images as latest or latest-rc, we can better utilize automated update tools such as Watchtower.

#### Here are a few key advantages:
1. **Simplicity**: By using tags like `latest` or `latest-rc`, we eliminate the need to manually update or specify the version number every time we want to pull, run or upgrade an image. By simply using `latest` or `latest-rc`, we can access the most recent stable or release candidate version respectively.
2. **Automated updates with Watchtower**: Watchtower is a tool that can periodically check for new versions of Docker images that we are using, based on the tags. If a new version of an image tagged latest or latest-rc is available, Watchtower automatically pulls and uses it, minimizing the need for manual intervention.

#### Here's how we do it with docker:

```sh
docker tag karrio/server:2024.2.rc7 karrio/server:latest-rc
docker push karrio/server:latest-rc
```

It's important to keep in mind that `latest` or `latest-rc` tags are simply conventions. They do not automatically point to the most recently pushed or built image unless we explicitly tag the image as `latest` or `latest-rc`. Therefore, we have the responsibility of tagging an image as `latest` or `latest-rc` during the build and management of Docker images.

**Comment by danh91** [comment]
Hey @jacobshilitz,

Thanks for the details. I just tagged and pushed `karrio/server:2024.2.rc7` as `latest-rc`
And `karrio/server:2024.2.rc4`  as `latest` since it has been stable for a while. 

I will tag the `2024.2` as `latest` later this week
